lib/config/presets/internal/workarounds.ts

```diff
@@ -118,4 +119,24 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  javaLTSVersions: {
+    description: 'Limit Java runtime versions to LTS releases',
+    packageRules: [
+      {
+        description:
+          'Limit Java runtime versions to LTS releases. To receive all major releases add `javaLTSVersions` to the `ignorePresets` array.',
+        matchDatasources: ['docker', 'adoptium-java'],
+        matchPackageNames: [
+          'eclipse-temurin',
+          'amazoncorretto',
+          'adoptopenjdk',
+          'openjdk',
+          'java',
+        ],
+        versioning:
+          'regex:^(?<major>\\d+)?(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?([\\._+](?<build>\\d+))?(-(?<compatibility>.*))?$',
+        allowedVersions: '/^(8|11|17|21|25|29)/',
```

Comment:
```suggestion
        allowedVersions: '/^(?:8|11|17|21|25|29)\./',
```
- require trailing dot after major otherwise 111.0 would also allowed
- use non-capturing groups

Reply:
Trailing dot or end of line? Eg in case there's Docker images with just "11"

Reply:
Good catch! I hope to never see Java 111.x haha

Reply:
updated suggestion